app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt

```diff
@@ -31,9 +44,9 @@ open class NewsAdapter(private val newsListener: OnItemClickListener, private va
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
         return when (viewType) {
-            0 -> NewsHolder(newsListener, LayoutInflater.from(parent.context).inflate(R.layout.news_holder, parent, false))
-            1 -> SourceHolder(sourceListener, LayoutInflater.from(parent.context).inflate(R.layout.source_holder, parent, false))
-            else -> NewsHolder(newsListener, LayoutInflater.from(parent.context).inflate(R.layout.news_holder, parent, false))
+            0 -> NewsHolder(newsListener, sourceListener, LayoutInflater.from(parent.context).inflate(R.layout.news_holder, parent, false))
```

Comment:
please make constants for the view types, don't use 1,2,3 here

Reply:
added companion object with 2 consts